src/weights.jl

```diff
@@ -291,3 +291,82 @@ end
 
 wmedian(v::RealVector, w::RealVector) = median(v, weights(w))
 wmedian{W<:Real}(v::RealVector, w::WeightVec{W}) = median(v, w)
+
+
+
+
+###### Weighted quantile #####
+# Variables m, g, low, high, correspond respectively to m, gamma, x[j], x[j+1] in the R help file for quantile
+
+function Base.quantile{W<:Real}(v::RealVector, p::RealVector, w::WeightVec{W})
+
+    isempty(v) && error("quantile of an empty array is undefined")
+    isempty(p) && throw(ArgumentError("empty quantile array"))
+
+    ppermute = sortperm(p)
+    p = p[ppermute]
+
+    # make sure the quantiles are in [0,1]
+    p = bound_quantiles(p)
+
+    if w.sum == 0
+        error("weight vector cannot sum to zero")
+    end
+    if length(v) != length(w)
+        error("data and weight vectors must be the same size. got $(length(v)) and $(length(w))")
+    end
+    for x in w.values
+        isnan(x) && error("weight vector cannot contain NaN entries")
+        x < 0 && error("weight vector cannot contain negative entries")
+    end
+    for x in v
+        isnan(x) && return x
+    end
+
+    lv = length(v)
+    lp = length(p)
+    m = 1 - p
+    index = (lv * p + m) * w.sum / lv
+    wt = w.values
+    vpermute = sortperm(v)
```

Comment:
It seems like there are too solutions

```julia
n = 1_000_000
a = rand(1:n, n)
b = rand(1:10, n)

@time sortperm([(a[i], b[i]) for i in 1:length(a)])
#elapsed time: 9.676994989 seconds   (287943584 bytes allocated, 16.39% gc time)
@time sortperm([(sub(a,i)[1], sub(b,i)[1]) for i in 1:length(a)], order= Base.Order.Lexicographic)
#elapsed time: 9.446759537 seconds (560247620 bytes allocated, 17.83% gc time)
```

I don't really understand the difference between these two solutions. Both are very slow (sorting `a` only takes 0.237).